packages/wrangler/src/dev.tsx

```diff
@@ -772,27 +774,31 @@ export async function startDev(args: StartDevOptions) {
 			});
 		}
 
-		if (args.experimentalAssets || config.experimental_assets) {
-			args.forceLocal = true;
-			if (
-				args.experimentalAssets &&
-				!existsSync(path.resolve(process.cwd(), args.experimentalAssets))
-			) {
+		const experimentalAssets = args.experimentalAssets
+			? { directory: args.experimentalAssets }
+			: config.experimental_assets;
+		if (experimentalAssets) {
+			const experimentalAssetsBasePath = getExperimentalAssetsBasePath(
+				config,
+				args.experimentalAssets
+			);
+			const resolvedExperimentalAssetsPath = path.resolve(
+				experimentalAssetsBasePath,
+				experimentalAssets.directory
+			);
+
+			if (!existsSync(resolvedExperimentalAssetsPath)) {
+				const sourceOfTruthMessage = args.experimentalAssets
+					? '"--experimental-assets" command line argument'
+					: '"experimental_assets.directory" field in your configuration file';
+
 				throw new UserError(
-					`The directory specified by the "--experimental-assets" command line argument does not exist:\n` +
-						`${path.resolve(process.cwd(), args.experimentalAssets)}`
+					`The directory specified by the ${sourceOfTruthMessage} does not exist:\n` +
+						`${resolvedExperimentalAssetsPath}`
```

Comment:
I feel a bit bad that this is a duplicate of the same code in the deploy command...

Reply:
yeah, I'm aware of the code duplication. I can abstract the user error logic into a separate fn if that helps. I don't have strong opinions either way, other than that we should not abstract all of this code block away, as that might become unreadable. Plus we might see some differences as we move further into the implementation.